internal/service/dynamodb/table_replica.go

```diff
@@ -417,6 +426,22 @@ func resourceTableReplicaUpdate(ctx context.Context, d *schema.ResourceData, met
 			}
 		}
 
+		if d.HasChange("deletion_protection_enabled") {
+			log.Printf("[DEBUG] Updating DynamoDB Table Replica deletion protection: %v", d.Get("deletion_protection_enabled").(bool))
+			_, err := conn.UpdateTable(ctx, &dynamodb.UpdateTableInput{
+				TableName:                 aws.String(tableName),
+				DeletionProtectionEnabled: aws.Bool(d.Get("deletion_protection_enabled").(bool)),
+			})
+			if err != nil {
+				return create.AppendDiagError(diags, names.DynamoDB, create.ErrActionUpdating, resNameTableReplica, d.Id(), err)
+			}
+			// Deletion protection change is reflected only on table status, and not in replicas status field
+			// There is certain lag after attr update and table status change so doing it with delay
+			if _, err := waitTableActiveAfterDeletionProtectionChange(ctx, conn, tableName, d.Timeout(schema.TimeoutUpdate)); err != nil {
```

Comment:
Curious whether we could still make this work with the existing waitTableActive function and an increased timeout, something for one of the core maintainers to check.